torchvision/transforms/functional_tensor.py

```diff
@@ -219,3 +220,40 @@ def ten_crop(img, size, vertical_flip=False):
 def _blend(img1, img2, ratio):
     bound = 1 if img1.dtype.is_floating_point else 255
     return (ratio * img1 + (1 - ratio) * img2).clamp(0, bound).to(img1.dtype)
+
+
+def resize(img, size, interpolation="bilinear"):
+    r"""Resize the input Tensor Image to the given size.
+
+    Args:
+        img (Tensor): Image to be resized.
+        size (sequence or int): Desired output size. If size is a sequence like
+            (h, w), the output size will be matched to this. If size is an int,
+            the smaller edge of the image will be matched to this number maintaing
+            the aspect ratio. i.e, if height > width, then image will be rescaled to
+            :math:`\left(\text{size} \times \frac{\text{height}}{\text{width}}, \text{size}\right)`
+        interpolation (string, optional): Desired interpolation ["bilinear", "nearest", "bicubic"]. Default is
+            ``bilinear``
+
+    Returns:
+         Tensor: Resized image Tensor.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    if isinstance(size, int):
+        w, h = img.shape[2], img.shape[1]
+        if (w <= h and w == size) or (h <= w and h == size):
+            return img
+        if w < h:
+            ow = size
+            oh = int(size * h / w)
+            out_img = Fn.interpolate(img.unsqueeze(0), size=(oh, ow), mode=interpolation)
+        else:
+            oh = size
+            ow = int(size * w / h)
+            out_img = Fn.interpolate(img.unsqueeze(0), size=(oh, ow), mode=interpolation)
+    else:
+        out_img = Fn.interpolate(img.unsqueeze(0), size=size, mode=interpolation)
```

Comment:
This is fine as is, but I think we can simplify a bit the code so that we only need to call `interpolate` in a single codepath.

A solution would be to create the `size` in the `int` case to be a tuple. Something like
```python
    if isinstance(size, int):
        w, h = img.shape[2], img.shape[1]
        if (w <= h and w == size) or (h <= w and h == size):
            return img
        if w < h:
            ow = size
            oh = int(size * h / w)
            size = (oh, ow)
    else:
            oh = size
            ow = int(size * w / h)
            size = (oh, ow)
    out_img = Fn.interpolate(..., size=size, mode=interpolation)
```